### PR TITLE
oSPARC-S4L M2 ready

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?

Milestone 2 ready:
- [x] No data tab for s4l product
- [x] ```App mode``` concept implemented:
  - No workbench, No logger, only full-screen
  - In dynamic services: Header and settings are hidden
  - If the pipeline is linear, ```App mode``` can be started and, by default, the entire pipeline will be part of the slideshow
- [x] Slideshow mode is persistent and an icon is shown in the study/template card
- [x] If a template is published in ```Guided mode```/ ```App mode```, the instance will be started in that mode
- [x] Make guided mode less buggy

```Guided mode``` vs ```App mode```:
Slideshow_guided-vs-app

Persistent and icons:
Slideshow_icons

```App mode``` template:
Slideshow_template_app

Add nodes from breadcrumbs:
Slideshow_addNodes

## Related issue/s
related to https://github.com/ITISFoundation/osparc-issues/issues/496
closes https://github.com/ITISFoundation/osparc-simcore/issues/2561
Disable share link for templates https://github.com/ITISFoundation/osparc-simcore/issues/2560

### CODE REVIEW

**Review by pcrespov — COMMENTED** [review]
👍

**Review by KZzizzle — APPROVED** [review]
looks all good to me - should it also have StS's review?
Just two comments:
* I'm not sure it makes sense to allow making a node invisible through the "quick edit" since none of the other operations are possible (e.g. making a node visible, rearranging nodes) so it might be confusing to only that one function available. However if the PO's wanted it like that, then ignore my comment =)
* When a new node is created via the guided mode "quick edit" and ports cannot be auto-connected perhaps there should be a warning?

**Comment by odeimaiz** [comment]
> * I'm not sure it makes sense to allow making a node invisible through the "quick edit" since none of the other operations are possible (e.g. making a node visible, rearranging nodes) so it might be confusing to only that one function available. However if the PO's wanted it like that, then ignore my comment =)

When you start the App Mode all nodes are added to the breadcrumbs, the App/GM mode designer might want to keep a computational service hidden (it will be run when the user "steps over")

**Review by pcrespov — APPROVED** [review]

**Comment by ignapas** [comment]
review in progress...

File: services/web/client/source/class/osparc/dashboard/Dashboard.js [hunk 1]
```diff
@@ -70,12 +70,14 @@ qx.Class.define("osparc.dashboard.Dashboard", {
     },
 
     __createMainViewLayout: function() {
-      [
+      const tabs = [
```
Comment (ignapas) [inline services/web/client/source/class/osparc/dashboard/Dashboard.js:73]: I like this syntax for better readability:
```javascript
const tabs = [
  { label: 'label', buildLayout: () => {} }
  // ...
]
// iterate
tabs.forEach(({ label, buildLayout }) => {
  const tabPage = new qx.ui.tabview.Page(label)
  const layout = buildLayout()
})
```

**Review by ignapas — APPROVED** [review]
imo indicator icons in dashboard items should have some sort of explanatory tooltip

File: services/web/client/source/class/osparc/dashboard/Dashboard.js (hunk 1 above)
Comment (odeimaiz) [reply services/web/client/source/class/osparc/dashboard/Dashboard.js]: thanks 👌

**Pull request merged**